commit-event-service/src/test/scala/ch/datascience/commiteventservice/events/categories/commitsync/eventgeneration/historytraversal/CommitInfoFinderSpec.scala

```diff
@@ -142,8 +143,117 @@ class CommitInfoFinderSpec extends AnyWordSpec with MockFactory with ExternalSer
     }
   }
 
-  private implicit val cs:    ContextShift[IO] = IO.contextShift(global)
-  private implicit val timer: Timer[IO]        = IO.timer(global)
+  "getMaybeCommitInfo" should {
+
+    "get commit info from the configured url " +
+      "and return some CommitInfo if OK returned with valid body - case with Personal Access Token" in new TestCase {
+
+        val accessToken = personalAccessTokens.generateOne
+
+        stubFor {
+          get(s"/api/v4/projects/$projectId/repository/commits/$commitId")
+            .withHeader("PRIVATE-TOKEN", equalTo(accessToken.value))
+            .willReturn(okJson(responseJson.toString()))
+        }
+
+        finder.getMaybeCommitInfo(projectId, commitId, Some(accessToken)).unsafeRunSync() shouldBe CommitInfo(
+          id = commitId,
+          message = commitMessage,
+          committedDate = committedDate,
+          author = author,
+          committer = committer,
+          parents = parents
```

Comment:
Are we sure if decoding works if there are no parents in the response?

Reply:
So this should be ok; if there are no parents gitlab returns an empty list, and the test covers this case.